include/arch/arm/arch/64/mode/object/structures.bf

```diff
@@ -190,7 +190,9 @@ block asid_map_none {
 
 block asid_map_vspace {
     field_high vspace_root          48
-    padding                         14
+    padding                         5
+    field hw_asid                   8
```

Comment:
But ASIDs are 16 bits for aarch64, not 8 bits.

Reply:
I believe in this specific part, these are 8bit VMIDs for stage 2 translation TLB.  Because there are 16 bit ASIDs in aarch64, there isn't a need for a mapping as the seL4 ASID can be directly used for the hardware ASID when the kernel is operating stage 1 page tables.  When the kernel is configured with CONFIG_ARM_HYPERVISOR_SUPPORT it only does stage 2 translation management, but now needs to map 16bit seL4 ASIDs to 8bit VMIDs and needs to track hw_asid assignment.

Reply:
VMIDs are independent of ASIDs, so I have no idea what you mean. If hw_asid is actually an VMID it should be named as such, or if some hackery is going on, well documented that it's actually an VMID for 64-bit ARM with hypervisor support. Are you saying that generic ASID mapping code is only used for VMIDs mapping by aarch64?

Reply:
These fields should be both renamed to VMIDs, and also #ifdef'd by CONFIG_ARM_HYPERVISOR_SUPPORT too.  The naming convention to use hw_asid is inherited from the 32bit hyp extensions.  We should probably look at fixing it for both if we can

Reply:
My reading is the hw_asid can be mapped to different things on different architectures, if I understand the author's intention correctly, without looking at other parts yet. 
On x86-64, it is mapped to PCID, VPID for EPT (not so sure about that part)? for armv7,v8 without HYP, it is mapped to asid, with HYP, it is mapped to vmid. RISCV also uses similar ASID and VMID naming. So the hw_asid might be a reasonable term to represent the underlying hardware mechanism without too many #ifdef in the code. Some early armv8 chips has 8-bit VMID instead of 16-bit.